src/lib/sticky-header/sticky-header.ts

```diff
@@ -0,0 +1,321 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Directive, Input,
+  OnDestroy, AfterViewInit, ElementRef, Optional} from '@angular/core';
+import {Platform} from '../core/platform';
+import {Scrollable} from '../core/overlay/scroll/scrollable';
+import {extendObject} from '../core/util/object-extend';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/debounceTime';
+import {isPositionStickySupported} from '../../cdk/platform/features';
+
+
+/**
+ * Directive that marks an element as a "sticky region", meant to contain exactly one sticky-header
+ * along with the content associated with that header. The sticky-header inside of the region will
+ * "stick" to the top of the scrolling container as long as this region is within the scrolling
+ * viewport.
+ *
+ * If a user does not explicitly define a sticky-region for a sticky-header, the direct
+ * parent node of the sticky-header will be used as the sticky-region.
+ */
+@Directive({
+  selector: '[cdkStickyRegion]',
+})
+export class CdkStickyRegion {
+  constructor(public readonly _elementRef: ElementRef) { }
+}
+
+
+/** Class applied when the header is "stuck" */
+const STICK_START_CLASS = 'cdk-sticky-header-start';
+
+/** Class applied when the header is not "stuck" */
+const STICK_END_CLASS = 'cdk-sticky-header-end';
+
+/**
+ * Set a debounce time which is used in debounce() function when adding event listeners.
+ * Set is as 5. Because if the DEBOUNCE_TIME is set as a too large number. The sticky effect
+ * during scroll will become vary strange and can not be scrolled smoothly.
+ */
+const DEBOUNCE_TIME: number = 5;
+
+/**
+ * Directive that marks an element as a sticky-header. Inside of a scrolling container (marked with
+ * cdkScrollable), this header will "stick" to the top of the scrolling viewport while its sticky
+ * region (see cdkStickyRegion) is in view.
+ */
+@Directive({
+  selector: '[cdkStickyHeader]',
+})
+export class CdkStickyHeader implements OnDestroy, AfterViewInit {
+
+  /**
+   * Set the sticky-header's z-index as 10 in default. Make it as an input
```

Comment:
:+1:  Changed the doc to` '/** z-index to be applied to the sticky header (default is 10). */'`